WeatherPortal/weatherapp/views.py

```diff
@@ -22,85 +57,145 @@ def register(request):
+                            'city': new_add_city,
+                            'lat': get_location_for_new_city[counter_cities]['lat'],
+                            'lon': get_location_for_new_city[counter_cities]['lon'],
+                            'country': get_location_for_new_city[counter_cities]['country'],
+                            'state': location_state,
+                            'number': counter_cities,
+                        }
+                        cities_locations.append(city_location)
             if error_msg_add_city_button:
                 message_for_user = error_msg_add_city_button
+
+    if add_new_location:
+        chosen_location = int(chosen_location)
+        last_add_location = request.session.get('last_add_location')
+        get_location_for_new_city = get_city_location(last_add_location)
+        lat_chosen_location = get_location_for_new_city[chosen_location]['lat']
+        lon_chosen_location = get_location_for_new_city[chosen_location]['lon']
+        if request.user.is_authenticated:
+            check_that_location_on_the_same_lat_lon_exist = Location.objects.filter(user=request.user,
+                                                                                    lat=lat_chosen_location,
+                                                                                    lon=lon_chosen_location).count()
+            if check_that_location_on_the_same_lat_lon_exist == 0:
+                get_weather_data = get_weather_data_by_location(lat_chosen_location, lon_chosen_location)
+                weather_for_new_location = {
+                    'location': last_add_location,
+                    'country': get_location_for_new_city[chosen_location]['country'],
+                    'location_number': chosen_location,
+                    'temperature': get_weather_data['current']['temp'],
+                    'wind': get_weather_data['current']['wind_speed'],
+                    'description': get_weather_data['current']['weather'][0]['description'],
+                    'icon': get_weather_data['current']['weather'][0]['icon'],
+                }
             else:
-                message_for_user = 'New City added successfully'
-
-        cities_for_user = City.objects.filter(user=request.user.id)
-        for city in cities_for_user:
-            get_weather_data = requests.get(url.format(city)).json()
-
-            weather_in_city = {
-                'city': city.name,
-                'temperature': get_weather_data['main']['temp'],
-                'wind': get_weather_data['wind']['speed'],
-                'description': get_weather_data['weather'][0]['description'],
-                'icon': get_weather_data['weather'][0]['icon'],
-            }
+                message_for_user = "Location already exist in your list"
+                add_new_location = False
+        else:
+            get_weather_data = get_weather_data_by_location(lat_chosen_location, lon_chosen_location)
```

Comment:
Ciężko się tutaj nie pogubić, za dużo tej ifologii, zdecydowanie trzeba to uprościć i podzielić na mniejsze części kodu, czyli na osobne funkcje

Reply:
Zmerguje, aby koledzy mogli działać, ale jak najłatwiej np. rozdzielić to do innej funkcji?:
```
    if add_new_location:
        chosen_location = int(chosen_location)
        last_add_location = request.session.get('last_add_location')
        get_location_for_new_city = get_city_location(last_add_location)
        lat_chosen_location = get_location_for_new_city[chosen_location]['lat']
        lon_chosen_location = get_location_for_new_city[chosen_location]['lon']
        if request.user.is_authenticated:
            check_that_location_on_the_same_lat_lon_exist = Location.objects.filter(user=request.user,
                                                                                    lat=lat_chosen_location,
                                                                                    lon=lon_chosen_location).count()
            if check_that_location_on_the_same_lat_lon_exist == 0:
                get_weather_data = get_weather_data_by_location(lat_chosen_location, lon_chosen_location)
                weather_for_new_location = {
                    'location': last_add_location,
                    'country': get_location_for_new_city[chosen_location]['country'],
                    'location_number': chosen_location,
                    'temperature': get_weather_data['current']['temp'],
                    'wind': get_weather_data['current']['wind_speed'],
                    'description': get_weather_data['current']['weather'][0]['description'],
                    'icon': get_weather_data['current']['weather'][0]['icon'],
                }
            else:
                message_for_user = "Location already exist in your list"
                add_new_location = False
        else:
            get_weather_data = get_weather_data_by_location(lat_chosen_location, lon_chosen_location)
```
przy return mam błąd.
Znalazłem coś takiego:
```
def HomeView(request):
  dict = func(request)
  ... # rest of code that is not common with other views
```
```
def func(request):
  #all repeated code
  #put all needed variables in a dictionary and return it
  dict = {'var1': 4, 'var2': "a String" ...}
  return dict
```
tylko to nie skróci znacznie funkcji